src/sensors/LaserRangeFinder.cpp

```diff
@@ -298,116 +300,34 @@ void LaserRangeFinder::setNumBeams(int num_beams) {
-        Vector3 dir = polarTo2D(a, 1);
-        ray_dirs_.push_back(dir);
-        angles_.push_back(a);
-        a += a_incr;
-    }
-
-    // Create a look-up table which translates slope to ray index. This way,
-    // we can later on calculate the approximate polar coordinates a point on the
-    // laser plane without using atan.
-
-    // Determine the needed resolution of the look-up table (TODO: make less ad-hoc)
-    slope_factor_ = 1 / tan(getAngleIncrement()) * 10;
-    int N = (int)slope_factor_ + 1;
-
-    for(unsigned int i = 0; i < 8; ++i)
-        slope_to_index_[i].resize(N);
-
-    // We divide the unit circle into 8 parts, split by 3 criteria:
-    // abs(x) < abs(y)
-    // x < 0
-    // y < 0
-
-    // This way we can always calculate a slope between 0 and 1 (by dividing the
-    // smallest coordinate by the biggest).
-    for(int j = 0; j < 8; ++j)
-    {
-        slope_to_index_[j].resize(N);
-        for(int k = 0; k < N; ++k)
-        {
-            // Calculate the slope corresponding to this table entry
-            double slope = (double)k / slope_factor_;
-
-            // Calculate a virtual (x, y) cartesian point which corresponds to this table entry
-            // based on the slope. Determine the part of the unit circle in which (x, y) lies
-            // base on j
-            double x, y;
-            if (j & 1)
-            {
-                // x_abs > y_abs
-                x = 1;
-                y = slope + 1e-16; // Necessary because of 0-check in getAngle
-            }
-            else
-            {
-                // x_abs < y_abs
-                x = slope + 1e-16; // Necessary because of 0-check in getAngle
-                y = 1;
-            }
-
-            if (!(j & 2))
-            {
-                // x < 0
-                x = -x;
-            }
-
-            if (!(j & 4))
-            {
-                // y < 0
-                y = -y;
-            }
-
-            // Calculate the angle of the virtual point (x, y)
-            double a = getAngle(x, y);
-            double a_temp = a - a_min_;
-            if (a_temp < 0)
-                a_temp += 2 * M_PI;
-
-            // Calculate the ray index based on the angle
-            slope_to_index_[j][k] = a_temp / (a_max_ - a_min_) * num_beams_ + 1;
-        }
+    for(uint i = 0; i < num_beams_; ++i) {
+        ray_dirs_[i] = polarTo2D(a, 1);
+        angles_[i] = a;
+        a += angle_incr_;
     }
 
-    i_half_circle_ = M_PI / getAngleIncrement();
+    i_half_circle_ = M_PI / angle_incr_;
 }
 
 double LaserRangeFinder::getAngleIncrement() const {
-    return (a_max_ - a_min_) / std::max(num_beams_ - 1, 1);
+    return angle_incr_;
 }
 
-int LaserRangeFinder::getAngleUpperIndex(double angle) const {
-    int i = (angle - a_min_) / (a_max_ - a_min_) * num_beams_ + 1;
-    return std::min(num_beams_, std::max(0, i));
+uint LaserRangeFinder::getAngleUpperIndex(double angle) const {
+    int i = (angle - a_min_) / angle_incr_ + 1;
```

Comment:
> Most important change is the change in getAngleUpperIndex.

So if I understand correctly this is the most important change? 

For clarity, this means you (effectively) changed 

> (a_max_ - a_min_) * num_beams_;

for 

> angle_incr_

 which is equal to

> (a_max_ - a_min_) / std::max<uint>(num_beams_ - 1, 1);

Reply:
True.

In the pass the angle increment was also `(angle_max - angle_min)/num_beans` which was incorrect. It has been changed to `(angle_max - angle_min)/(num_beans-1)` a while back. And this function has been missed.